mycore-ocfl/src/main/java/org/mycore/ocfl/user/MCROCFLXMLUserManager.java

```diff
@@ -64,22 +65,35 @@ public class MCROCFLXMLUserManager {
 
     private static final String IGNORING_TRANSIENT_USER = "Got TransientUser, ignoring...";
 
-    private final OcflRepository repository;
+    private OcflRepository repository;
+
+    @MCRProperty(name = "OCFL.Repository", required = true)
+    public String repositoryKey;
 
     /**
-     * Initializes the UserManager with the in the config defined repository.
+     * Initializes the UserManager
      */
     public MCROCFLXMLUserManager() {
-        this.repository = MCROCFLRepositoryProvider.getRepository(
-            MCRConfiguration2.getStringOrThrow("MCR.Users.Manager.Repository"));
+
     }
 
     /**
      * Initializes the UserManager with the given repositoryKey.
      * @param repositoryKey the ID for the repository to be used
      */
     public MCROCFLXMLUserManager(String repositoryKey) {
-        this.repository = MCROCFLRepositoryProvider.getRepository(repositoryKey);
+        this.repositoryKey = repositoryKey;
+        repository = MCROCFLRepositoryProvider.getRepository(repositoryKey);
+    }
+
+    @MCRPostConstruction
+    @SuppressWarnings("PMD")
+    public void init(String property) {
+        repository = MCROCFLRepositoryProvider.getRepository(repositoryKey);
+    }
+
+    public OcflRepository getRepository() {
+        return repository;
```

Comment:
too much code and silly init() mehod with unused parameter. Again double initialization.